pages-e2e/features/simpleAppSsrRoutes/navigation.test.ts

```diff
@@ -0,0 +1,72 @@
+import { describe, it } from 'vitest';
+
+describe('Simple App Static SPA navigation', () => {
+	it('should soft navigate between static routes', async ({ expect }) => {
+		const page = await BROWSER.newPage();
+
+		const requestUrls: string[] = [];
+		await page.route('**/*', route => {
+			requestUrls.push(route.request().url());
+			route.continue();
+		});
+
+		await page.goto(`${DEPLOYMENT_URL}/ssr-navigation`);
+
+		expect(
+			await page
+				.locator('h1', { hasText: 'Server Side Rendered Navigation' })
+				.isVisible(),
+		).toBe(true);
+		expect(
+			await page
+				.locator('h2', { hasText: 'Server Side Rendered Index' })
+				.isVisible(),
+		).toBe(true);
+
+		await page.locator('a', { hasText: 'to page A' }).click();
+
+		await page
+			.locator('h2', { hasText: 'Server Side Rendered Page A' })
+			.waitFor();
+		expect(
+			await page
+				.locator('h2', { hasText: 'Server Side Rendered Page A' })
+				.isVisible(),
+		).toBe(true);
```

Comment:
This is an ugly hack workaround, we should not need the `.waitFor` call but only the expect (or similar alternatives)

as playwright should do some some auto-waiting and reties
[![Screenshot 2023-07-17 at 16 38 47](https://github.com/cloudflare/next-on-pages/assets/61631103/04ddc9b2-03b5-4002-b7a1-4bc4d5134b14)](https://playwright.dev/docs/api/class-locator)
not forcing developers do manually wait for elements to appear in the dom

___

> **Note**
> The worst parse is that `waitFor` throws a generic `TimeoutError`, I think I need to add a utility for this sort of check and there catch that and rethrow it with a more clear message.

Reply:
I've improved this with an `assertVisible` utility

___

### Before and after

#### Code
![Screenshot 2023-07-18 at 12 21 03](https://github.com/cloudflare/next-on-pages/assets/61631103/809d803d-23ca-4da5-88f2-1252be85cbda)

#### Failure message

![Screenshot 2023-07-18 at 12 27 27](https://github.com/cloudflare/next-on-pages/assets/61631103/2bbe0014-350e-4d1a-bdcd-2abb0f4bf611)


![Screenshot 2023-07-18 at 12 03 57](https://github.com/cloudflare/next-on-pages/assets/61631103/6cf06005-609b-4137-bf1a-35947df6bb94)